Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,342 @@
+  // Fallback to simple quad when there are no obstacles around.
+  if(raycastResult.length === 0) {
+    if(!this._defaultVertexBuffer) {
+      this._defaultVertexBuffer = new Float32Array(8);
+    }
+    if(!this._defaultIndexBuffer) {
+      this._defaultIndexBuffer = new Uint16Array([
+        0, 1, 2, 0, 2, 3
+      ]);
+    }
+
+    this._defaultVertexBuffer[0] = this._object.x - this._radius;
+    this._defaultVertexBuffer[1] = this._object.y + this._radius;
+    this._defaultVertexBuffer[2] = this._object.x + this._radius;
+    this._defaultVertexBuffer[3] = this._object.y + this._radius;
+    this._defaultVertexBuffer[4] = this._object.x + this._radius;
+    this._defaultVertexBuffer[5] = this._object.y - this._radius;
+    this._defaultVertexBuffer[6] = this._object.x - this._radius;
+    this._defaultVertexBuffer[7] = this._object.y - this._radius;
+
+    this._light.geometry
+      .getBuffer('aVertexPosition')
+      .update(this._defaultVertexBuffer);
+    this._light.geometry.getIndex().update(this._defaultIndexBuffer);
+    return;
+  }
+
+  var raycastResultLength = raycastResult.length;
+
+  var isSubArrayUsed = false;
+  var vertexBufferSubArray = null;
+  var indexBufferSubArray = null;
+
+  if (this._vertexBuffer.length > 2 * raycastResultLength + 2) {
+    if (this._vertexBuffer.length < 4 * raycastResultLength + 4) {
+      isSubArrayUsed = true;
+      vertexBufferSubArray = this._vertexBuffer.subarray(
+        0,
+        2 * raycastResultLength + 2
+      );
+      indexBufferSubArray = this._indexBuffer.subarray(
+        0,
+        3 * raycastResultLength
+      );
+    } else {
+      this._vertexBuffer = new Float32Array(2 * raycastResultLength + 2);
+      this._indexBuffer = new Uint16Array(3 * raycastResultLength);
+    }
+  }
+
+  if (this._vertexBuffer.length < 2 * raycastResultLength + 2) {
+    this._vertexBuffer = new Float32Array(2 * raycastResultLength + 2);
+    this._indexBuffer = new Uint16Array(3 * raycastResultLength);
+  }
+
+  this._vertexBuffer[0] = this._object.x;
+  this._vertexBuffer[1] = this._object.y;
+
+  for (var i = 2; i < 2 * raycastResultLength + 2; i += 2) {
+    this._vertexBuffer[i] = raycastResult[i / 2 - 1][0];
+    this._vertexBuffer[i + 1] = raycastResult[i / 2 - 1][1];
+  }
+
+  for (var i = 0; i < 3 * raycastResultLength; i += 3) {
+    this._indexBuffer[i] = 0;
+    this._indexBuffer[i + 1] = i / 3 + 1;
+    if (i / 3 + 1 !== raycastResultLength) this._indexBuffer[i + 2] = i / 3 + 2;
+    else this._indexBuffer[i + 2] = 1;
+  }
+
+  if (!isSubArrayUsed) {
+    this._light.geometry
+      .getBuffer('aVertexPosition')
+      .update(this._vertexBuffer);
+    this._light.geometry.getIndex().update(this._indexBuffer);
+  } else {
+    this._light.geometry
+      .getBuffer('aVertexPosition')
+      .update(vertexBufferSubArray);
+    this._light.geometry.getIndex().update(indexBufferSubArray);
+  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
```

Comment:
Let's find a name that explain what is returned/done in this function, and that it's fairly expensive. Something like `computeLightVertices`? (compute = "expensive")